src/main/java/com/google/growpod/controllers/GardenDao.java

```diff
@@ -118,4 +119,107 @@ public Garden getGardenById(String id) {
 
     return plantList;
   }
+
+  /**
+   * Adds a plant to a garden's plant list.
+   *
+   * @param gardenId the garden to add the plant to
+   * @param plant the plant object
+   * @return The plant's key
+   */
+  public String addToGardenPlantList(String gardenId, Plant plant) {
+    // Add plant to plant list first
+    plant.setId("1"); // Dummy key to make .toEntity(datastoreInstance) work.
+    KeyFactory keyFactory = datastore.newKeyFactory().setKind("Plant");
+    IncompleteKey incompleteKey = keyFactory.newKey();
+
+    Key key = datastore.allocateId(incompleteKey);
+
+    Entity newEntity = Entity.newBuilder(key, plant.toEntity(datastoreInstance)).build();
+    datastore.add(newEntity);
+
+    String plantId = Long.toString(newEntity.getKey().getId());
+
+    // Then add relation
+    keyFactory = datastore.newKeyFactory().setKind("ContainsPlant");
+    incompleteKey = keyFactory.newKey();
+
+    key = datastore.allocateId(incompleteKey);
+    ContainsPlant relation = new ContainsPlant("1", gardenId, plantId);
+
+    newEntity = Entity.newBuilder(key, relation.toEntity(datastoreInstance)).build();
+    datastore.add(newEntity);
+    return plantId;
+  }
+
+  /**
+   * Deletes a user id from the garden's user list.
+   *
+   * @param gardenId the garden's id
+   * @param userId the user's id
+   * @return whether the query was successful.
+   */
+  public boolean deleteFromGardenUserList(String gardenId, String userId) {
```

Comment:
nit:
I find this name slightly confusing. What do you think of `removeUserFromGarden` or the more terse `deleteUser`. It becomes clear from context that it is from a garden, i.e.:

```java
gardenDao.deleteUser(gardenId, userId)
```

Similar comment with `addToGardenPlantList`.  Could become:
`gardenDao.addPlant(gardenId, plant)`

Reply:
Will update.